django/http/request.py

```diff
@@ -572,3 +578,38 @@ def validate_host(host, allowed_hosts):
     Return ``True`` for a valid host, ``False`` otherwise.
     """
     return any(pattern == '*' or is_same_domain(host, pattern) for pattern in allowed_hosts)
+
+
+class EnvironHeaders(CaseInsensitiveMapping):
+    IGNORE_EXCEPTIONS = {'HTTP_CONTENT_TYPE', 'HTTP_CONTENT_LENGTH'}
```

Comment:
I think this deserves a comment. Why are these headers "ignored" (whatever that means). Why are the headers below "special unchanged" (whatever that means).

Reply:
Maybe just reference that these don't get the prefix as specified in https://www.python.org/dev/peps/pep-0333/

I understand why `SPECIAL_UNCHANGED_HEADERS` is needed but I don't understand why `IGNORE_EXCEPTIONS` is needed. Do we ever get incoming `HTTP_CONTENT_TYPE` keys?

Reply:
I remember I did some research about the possible variations of multiple WSGI servers and some users noticed that some servers were not honoring PEP-333 and included `HTTP_CONTENT_TYPE`. 

I found that Python's [`wsgiref.validate`](https://docs.python.org/3.6/library/wsgiref.html) module [raises an exception](https://github.com/python/cpython/blob/master/Lib/wsgiref/validate.py#L310) if ever found.

Options:

a) leave it as it is: A robust alternative. If someone is using a poorly designed wsgi server, it'll still work. Maybe issue a warning.
b) Remove it all together. Assume that wsgi implementations will honor PEP 333.
c) Make it fail if ever found one (raise an exception). Similar to what `wsgiref.validate` is doing. This is kind of redundant, i don't like it TBH.

Reply:
Ok, yeah it looks like uwsgi might have, at least at some point, both CONTENT_TYPE and HTTP_CONTENT_TYPE.  So, yeah, I think ignoring HTTP_CONTENT_TYPE and trusting only CONTENT_TYPE is a good idea (a).

Reply:
@timgraham all points of your review addressed and I added the comments on these variables, just in case.

Reply:
I don't think Django has existing workarounds for this issue, e.g. https://github.com/django/django/blob/f021c110d02fd7ca32ae56f511b46e5d138b6c73/django/core/handlers/wsgi.py#L83
I think it adds necessary complexity.

Reply:
@timgraham 👍